flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -399,6 +413,40 @@ public static ResourceRequirements getResourceRequirements(
         return resourceRequirementsBuilder.build();
     }
 
+    /**
+     * Calculates the final limit value for a resource based on the new value, limit factor, and
+     * existing resource requirements.
+     *
+     * @param newValue The new value for the resource.
+     * @param limitFactor The limit factor for the resource.
+     * @param resourceRequirements Existing resource requirements.
+     * @param resourceName The name of the resource.
+     * @return The final limit value for the resource.
+     */
+    private static double getLimit(
+            double newValue,
+            double limitFactor,
+            ResourceRequirements resourceRequirements,
+            String resourceName) {
+        Map<String, Quantity> limits = resourceRequirements.getLimits();
+        double limit = newValue * limitFactor;
+        if (limits != null) {
+            Quantity quantity = limits.get(resourceName);
+            if (quantity != null) {
+                try {
+                    return Math.max(Double.parseDouble(quantity.getAmount()), limit);
```

Comment:
We should always log on info level that we are using the limit from the podTemplate instead of the config to avoid confusion